src/core/egg-herbie.rkt

```diff
@@ -125,39 +124,86 @@
       [(? symbol? x) (egraph_add_node ptr (symbol->string x) 0-vec root?)]
       [(? number? n) (egraph_add_node ptr (number->string n) 0-vec root?)]))
 
+  ; The function recurses on spec
+  (define (batch-parse-approx batch)
+    (batch-replace batch
+                   (lambda (node)
+                     (match node
+                       [(approx spec impl) (list '$approx spec impl)]
+                       [_ node]))))
+
+  (set-batch-roots! batch roots) ; make sure that we work with the right roots
```

Comment:
Since `batch-replace` and `remove-zombie-nodes` depend on `batch-roots`, we want to make sure that the roots are set to be the right ones